### PR TITLE
feature: Support Hetzner cloud provider in Helm chart

### PR DESCRIPTION
**Pull request by NiklasRosenstein**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Adds support for the Hetzner cloud provider in the cluster-autoscaler Helm chart.

#### Which issue(s) this PR fixes:

n/a

#### Special notes for your reviewer:
n/a

#### Does this PR introduce a user-facing change?

NONE

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

n/a

### CODE REVIEW

**Comment by Shubham82** [comment]
Hi @NiklasRosenstein
Please address the failed Test Cases i.e. `Lint and Test Charts / Helm chart (pull_request)` and `Lint and Test Charts / Helm Docs (pull_request)`, that are required to be passed.

JFYI: you have to install [Pre-commit hooks](https://github.com/kubernetes/autoscaler/tree/master/charts#install-pre-commit-binary) before working in the Charts.

**Comment by Shubham82** [comment]
The `Lint and Test Charts / Helm chart (pull_request)` test case failed due to not doing a bump version:
`✖︎ cluster-autoscaler => (version: "9.34.1", path: "charts/cluster-autoscaler") > Chart version not ok. Needs a version bump! `

So you need to bump the chart version to `9.34.2` as it can't publish a new chart otherwise.

**Comment by Shubham82** [comment]
The `Lint and Test Charts / Helm Docs (pull_request)` test case failed because you directly did the changes in the [README.md](https://github.com/kubernetes/autoscaler/blob/master/charts/cluster-autoscaler/README.md), we use [Pre-commit hooks](https://github.com/kubernetes/autoscaler/tree/master/charts#pre-commit-hooks) to update `README.md` file.

So The  README.md is generated by helm-docs from the `README.md.gotmpl` file, which is one of our [pre-commit hooks](https://github.com/kubernetes/autoscaler/tree/master/charts#pre-commit-hooks). So if you want to make changes to the README other than those in the values.yaml file, you will need to make changes in the `README.md.gotmpl` file.

**Comment by NiklasRosenstein** [comment]
Thanks @Shubham82, and sorry for not reading up on the contribution guidelines!

Just want to mention as a side note that your CI is using helm-docs 1.3.0 which doesn't seem to support an extra empty line in the `values.yaml` comments:

```diff
diff --git a/charts/cluster-autoscaler/values.yaml b/charts/cluster-autoscaler/values.yaml
index 5503dda51..b838afbf1 100644
--- a/charts/cluster-autoscaler/values.yaml
+++ b/charts/cluster-autoscaler/values.yaml
@@ -36,7 +36,6 @@ autoDiscovery:
 #   maxSize: 2<br />
 #   minSize: 1
 # </pre>
-#
 # For Hetzner Cloud, the `instanceType` and `region` keys are also required.
 # <pre>
 # - name: mypool<br />
```

The version I got with `brew install norwoodj/tap/helm-docs` today is 1.12.0 which does seem to support it and it proved a minor hiccup to get the CI to pass. I was able to run the older version with

```
$ docker run --rm -it --workdir /github/workspace -v $PWD:/github/workspace jnorwood/helm-docs:v1.3.0
INFO[2024-02-06T22:29:00Z] Found Chart directories [charts/cluster-autoscaler] 
INFO[2024-02-06T22:29:00Z] Generating README Documentation for chart /github/workspace/charts/cluster-autoscaler 
```

File: charts/cluster-autoscaler/Chart.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ name: cluster-autoscaler
 sources:
   - https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
 type: application
-version: 9.35.0
+version: 9.36.0
```
Comment (Shubham82) [inline charts/cluster-autoscaler/Chart.yaml:14]:
```suggestion
version: 9.35.1
```

**Review by Shubham82 — CHANGES_REQUESTED** [review]

File: charts/cluster-autoscaler/Chart.yaml (hunk 1 above)
Comment (Shubham82) [reply charts/cluster-autoscaler/Chart.yaml:14]: It should be the patch version, not the minor version.

Comment (gjtempleton) [reply charts/cluster-autoscaler/Chart.yaml]: Disagree with this, we're adding new functionality, this should be a minor version bump.

Comment (Shubham82) [reply charts/cluster-autoscaler/Chart.yaml]: Hi @gjtempleton, IIRC previously while adding support for exoscale provider, we updated the patch version.
Here is the PR: #6249, so I did the above review. But if you think it is good to have a minor version then I have no problem with it.
Thanks for the above information.

Comment (Shubham82) [reply charts/cluster-autoscaler/Chart.yaml]: @NiklasRosenstein could you please revert the version to `9.36.0`, as suggested by @gjtempleton.

Comment (NiklasRosenstein) [reply charts/cluster-autoscaler/Chart.yaml:14]: Done!

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve

**Pull request merged**